tests/v2/validation/airgap/airgap.go

```diff
@@ -0,0 +1,33 @@
+package airgap
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/registries"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	corralPackageAirgapCustomClusterName = "airgapCustomCluster"
+	corralBastionIP                      = "bastion_ip"
+	corralRegistryIP                     = "registry_ip"
+	corralRegistryFQDN                   = "registry_fqdn"
+	logMessageKubernetesVersion          = "Validating the current version is the upgraded one"
+)
+
+func checkPodStatusImageSource(t *testing.T, client *rancher.Client, clusterName, registryFQDN string) {
```

Comment:
nit; I argue that this would make more sense in a `registry/registry.go` file instead, since technically this is a use case for registries that airgap uses.

Reply:
As a framework extension in registry/registry.go? @izaac what do you think? Currently in your tests you have them as separate test validations/cases. If you were to somewhat rework your tests as RKE1, RKE2, K3S I would agree with Caleb, otherwise I would keep them separate.

Reply:
Those validations are identical for all cluster types and registries setup.

Reply:
Yeah, but in your tests you have them as separate test cases. So in your case would you use this extension for your validations? Also since 2.6 is EOM should this be back ported?

Reply:
I think it's safe to consolidate them as long as the failure indicates what failed, status or prefix IMO